indigo_app/forms.py

```diff
@@ -114,6 +119,26 @@ def save_publication_document(self):
             pub_doc.mime_type = self.cleaned_data['publication_document_mime_type']
             pub_doc.save()
 
+    def save_commencement(self):
+        # get existing main commencement object if there is one
+        main_commencement = self.instance.main_commencement()
+
+        # if work has been edited to not commence, delete existing main commencement object
+        if main_commencement and not self.instance.commenced:
+            main_commencement.delete()
+
+        # otherwise, amend the existing one (or create a new one) with the work / date given in the form
+        else:
+            if 'commencement_date' in self.changed_data or 'commencing_work' in self.changed_data:
+                if not main_commencement:
+                    all_provisions = self.instance.all_provisions()
+                    main_commencement = Commencement(commenced_work=self.instance, main=True, provisions=all_provisions)
+
+                main_commencement.commencing_work = self.cleaned_data.get('commencing_work')
```

Comment:
Just sanity check that it does what we expect if one of `commencing_work` or `commencement_date` are None.